R/clean_ProteinProspector.R

```diff
@@ -0,0 +1,45 @@
+#' Clean raw Protein Prospector data
+#' @param msstats_object object that inherits from MSstatsInputFiles class.
+#' @return data.table
+#' @keywords internal
+#' @noRd
+.cleanRawProteinProspector = function(msstats_object) {
+    protein_prospector_input = getInputFile(msstats_object, "input")
+    protein_prospector_input = 
+        data.table::as.data.table(protein_prospector_input)
+    channels = .getChannelColumns(
+        colnames(protein_prospector_input), "Int")
+    req_cols = c('AccX', 'z', 'DBPeptide', 'Fraction', channels)
+    protein_prospector_input = protein_prospector_input[, req_cols, with = FALSE]
+    data.table::setnames(
+        protein_prospector_input, 
+        c("AccX", "DBPeptide", "z", "Fraction"),
+        c("ProteinName", "PeptideSequence", "PrecursorCharge", "Run"), 
+        skip_absent = TRUE)
+    protein_prospector_input$PSM = paste(
+        protein_prospector_input$PeptideSequence, 
+        protein_prospector_input$PrecursorCharge,
+        1:nrow(protein_prospector_input), sep = "_"
+    )
+    
+    protein_prospector_input = melt(protein_prospector_input, 
+                                    measure.vars = channels, 
+                                    id.vars = setdiff(
+                                        colnames(protein_prospector_input), 
+                                        channels
+                                    ),
+                                    variable.name = "Channel", 
+                                    value.name = "Intensity"
+                                )
+    protein_prospector_input$Channel = .standardizeColnames(
+        protein_prospector_input$Channel
+    )
+    protein_prospector_input$Intensity = ifelse(
+        protein_prospector_input$Intensity == 0, 
```

Comment:
Is it exactly 0 or should we use abs(Intensity) < 1e-8 for example?
Also the [, Intensity := ifelse(....)] would be better.
Moreover, there might be a fast ifelse in data.table

Reply:
Update: MSstatsBalancedDesign does the same. It seems that input files should have exact 0s as missing values. I wonder if there's any case when this fails. Probably not

Reply:
Oh yeah you're right, balanced design does handle this case.  I'll remove this code here.

Reply:
I'll need to double check with the end user on what criteria warrants a missing value. I also looked through the larger input dataset provided and saw some negative numbers too, which is interesting.

Reply:
Ok I confirmed 0s and negative values are missing values